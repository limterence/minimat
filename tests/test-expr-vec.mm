/* test-expr-vec: vector assignment, subselect, definition, colon expression */
constant sequence x = [-99];     /* sequence constant definition */
int main() {
  sequence u;
  sequence v;
  sequence w;
  v = [1, 2, 3, 4];              /* construct sequence literal */
  printseq(v);
  printseq([]);                  /* empty sequence */
  v = [10, v, v, 11];            /* augment sequence */
  printseq(v);
  v = [101::105, 110:-1:106];    /* colon expression */
  printseq(v);
  printseq(v[2]);                /* subselect vector */
  u = [2, 5, 6];
  printseq(v[u]);
  v[0] = [200];                  /* subassign vector */
  printseq(v);
  v = u;                         /* assignment (copy values) */
  v[0] = [x];
  printseq(u);
  printseq(v);
  printseq(u[1] = v[1] = [99]);  /* chain assignment */
  printseq(u = v = [88, 888]);
}
