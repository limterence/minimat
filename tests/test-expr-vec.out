[4 int]
1 2 3 4 
[0 int]
[10 int]
10 1 2 3 4 1 2 3 4 11 
[10 int]
101 102 103 104 105 110 109 108 107 106 
[1 int]
103 
[3 int]
103 110 109 
[10 int]
200 102 103 104 105 110 109 108 107 106 
[3 int]
2 5 6 
[3 int]
-99 5 6 
[1 int]
99 
[2 int]
88 888 
